### DISCUSSION TITLE
How to get suspense option safety in useErrorBoundary?

### DISCUSSION THREAD

**Discussion by coo-kiee** [TanStack/query#4992] [Q&A]
I want to return dynamic boolean value in defaulOptions in useErrorBoundary
I can see it has suspense property in console.log, but type is not have it

So, i detour type like below
```
const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
            suspense: false,
            staleTime: 1000 * 5,
            cacheTime: 0,
            refetchOnWindowFocus: false,
            refetchOnMount: false,
            refetchOnReconnect: false,
            useErrorBoundary(error, query) {
                const queryOption = query.options as unknown as Record<string, unknown>
                console.log('Error ReactQuery : ', query.queryKey)
                console.log('Error Message : ', error)
                return queryOption.suspense ? true : false
            },
        },
    },
})
```

But, I think it is unsafety in feature,
So, how can i get suspense options safety?

**Comment by TkDodo** [comment]
not sure what you're trying to do but `useErrorBoundary` defaults to `true` if you use `suspense` and defaults to `false` otherwise, so your code here wouldn't be needed, as it's the default behaviour.

Other than that, there is no way to get that information because `suspense` is an observer level property, not a cache level one. So the same query could be `suspense:true` for one observer and `false` for another.
